src/vcpkg/registries.cpp

```diff
@@ -722,30 +719,35 @@ namespace
 
     // } RegistryImplementation
 
+    LocalizedString format_version_git_entry_missing(StringView port_name,
+                                                     const Version& expected_version,
+                                                     const std::vector<Version>& versions)
+    {
+        auto error_msg =
+            msg::format_error(msgVersionGitEntryMissing, msg::package_name = port_name, msg::version = expected_version)
+                .append_raw('\n');
+        for (auto&& version : versions)
+        {
+            error_msg.append_indent().append_raw(version.to_string()).append_raw('\n');
+        }
+
+        error_msg.append(msgVersionIncomparable4);
+        return error_msg;
+    }
+
     // { RegistryEntry
 
     // { BuiltinRegistryEntry::RegistryEntry
-    ExpectedS<PathAndLocation> BuiltinGitRegistryEntry::get_version(const Version& version) const
+    ExpectedL<PathAndLocation> BuiltinGitRegistryEntry::get_version(const Version& version) const
     {
         auto it = std::find(port_versions.begin(), port_versions.end(), version);
         if (it == port_versions.end())
         {
-            return {
-                Strings::concat(
-                    "Error: No version entry for ",
-                    port_name,
-                    " at version ",
-                    version,
-                    ". This may be fixed by updating vcpkg to the latest master via `git "
-                    "pull`.\nAvailable versions:\n",
-                    Strings::join("", port_versions, [](const Version& v) { return Strings::concat("    ", v, "\n"); }),
-                    "\nSee `vcpkg help versioning` for more information."),
-                expected_right_tag};
+            return format_version_git_entry_missing(port_name, version, port_versions);
```

Comment:
For the `BuiltinGitRegistry`, we should suggest updating the vcpkg instance. This registry type requires user action to make new versions available.

Reply:
For once I like the upgrade message! Agreed!